modular_citadel/code/modules/reagents/reagent container/hypospraymkii.dm

```diff
@@ -2,105 +2,105 @@
 	icon_state = "cmo2"
-	ignore_flags = 1
-	desc = "The Chief Medical Officer's hypospray is identically functional to the base model, excepting that it can take larger vials in addition to regular sized. It is also able to penetrate harder materials and deliver more reagents per spray."
+	desc = "The Deluxe Hypospray can take larger-size vials. It also acts faster and delivers more reagents per spray."
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/CMO
-	possible_transfer_amounts = list(5,10,15,30,60) //cmo hypo should be able to dump lots into it
-
-/obj/item/reagent_containers/hypospray/mkii/CMO/combat
-	name = "Combat Hypospray mk. II"
-	desc = "A hypospray designed for in-combat situations."
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/combat
-
-/obj/item/reagent_containers/hypospray/mkii/Initialize()
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/CMO
+	inject_wait = 20
+	spray_wait = 20
+
+/obj/item/hypospray/mkii/CMO/combat
+	name = "combat hypospray mk.II"
+	desc = "A combat-ready deluxe hypospray that acts almost instantly. It can be tactically reloaded by using a vial on it."
+	icon_state = "combat2"
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/combat
+	inject_wait = 0
+	spray_wait = 0
+	quickload = TRUE
+
+/obj/item/hypospray/mkii/Initialize()
 	. = ..()
 	if(!spawnwithvial)
 		update_icon()
 		return
-	if(!start_vial)
-		start_vial = new loaded_vial(src)
-		vial = start_vial
-		reagents.maximum_volume = vial.volume
-		vial.reagents.trans_to(src, vial.reagents.total_volume)
+	if(start_vial)
+		vial = new start_vial
 	update_icon()
 
-/obj/item/reagent_containers/hypospray/mkii/update_icon()
+/obj/item/hypospray/mkii/update_icon()
 	..()
 	icon_state = "[initial(icon_state)][vial ? "" : "-e"]"
 	if(ismob(loc))
 		var/mob/M = loc
 		M.update_inv_hands()
 	return
 
-/obj/item/reagent_containers/hypospray/mkii/examine(mob/user)
+/obj/item/hypospray/mkii/examine(mob/user)
 	. = ..()
+	to_chat(user, "[vial] has [vial.reagents.total_volume]u remaining.")
 	to_chat(user, "[src] is set to [mode ? "Inject" : "Spray"] contents on application.")
 
-/obj/item/reagent_containers/hypospray/mkii/proc/unload_hypo(obj/item/I, mob/user)
+/obj/item/hypospray/mkii/proc/unload_hypo(obj/item/I, mob/user)
 	if((istype(I, /obj/item/reagent_containers/glass/bottle/vial)))
 		var/obj/item/reagent_containers/glass/bottle/vial/V = I
-		reagents.trans_to(V, reagents.total_volume)
-		reagents.maximum_volume = 0
 		V.forceMove(user.loc)
 		user.put_in_hands(V)
-		to_chat(user, "<span class='notice'>You remove the vial from the [src].</span>")
+		to_chat(user, "<span class='notice'>You remove [vial] from [src].</span>")
```

Comment:
the vial is okay for a description term here, because vials can be renamed by the user.